### ISSUE TITLE
Websocket support

### ISSUE THREAD

**Issue by gbrlmtrz** [usebruno/bruno#316]
Hello, great project!

Are there any plans to support websocket requests in the near future?

**Comment by PalmDevs** [comment]
Yes please! This looks like a good replacement to Insomnia.

**Comment by xuliang5114923** [comment]
Looking forward to this as well +1

**Comment by neochne** [comment]
Looking forward to this as well +1

**Comment by zastlx** [comment]
+1

**Comment by Zeeex** [comment]
![image](https://github.com/usebruno/bruno/assets/16362320/c7a30c88-3415-427d-94ee-5fd55a45b64e)
Don't just restrict it behind a paywall, please.

**Comment by tfrancois** [comment]
+1.  Need websocket and socket.io support please.  ASAP.

**Comment by bartekpacia** [comment]
I'm worried that this feature is gonna be what Cloud Sync was to Postman and Insomnia.

Work you've done with Bruno is admirable, and this project is undeniably successful (and I love and use it) -- but if the goal is to paywall this feature*, this makes me worried how long the goals of maintainers of Bruno will stay aligned with the best interest of us, users.

*A feature that could be well implemented in the core. Or maybe even contributed – what will you with such a contribution then? Reject it? Are we, programmers, doomed to change HTTP GUI client every few years, because they become enshittified? At least this one is open-source, so can be forked.

**Comment by asjur** [comment]
I already bought Bruno as a one-time fee for this is more than acceptable. However, I don't like how silent the project has been on this feature, updates have been requested in several tickets with seemingly none answered.

**Comment by helloanoop** [comment]
Hey @asjur !

Websocket feature is expected to be released towards the end of this year.

**Comment by neysofu** [comment]
Any news @helloanoop? :pray:

**Comment by amunrarara** [comment]
Bring on the WebSocket! +1

**Comment by JorbFreire** [comment]
The plans to build it was discontinued? Could not find any mention about golden edition or socket features anywhere on pricing page.

**Comment by Its-treason** [comment]
> The plans to build it was discontinued? Could not find any mention about golden edition or socket features anywhere on pricing page.

See: https://docs.google.com/document/d/1ODgyIAw9bVAL-Z5K8FjdwQ_SEApuaxJFp11ZkrO8-Qg 
TL;DR: Websocket support will be available in the open source edition in the future.

**Comment by DannyvanHolten** [comment]
> Hey @asjur !
> 
> Websocket feature is expected to be released towards the end of this year.

Any update on this?

**Comment by Its-treason** [comment]
> > Hey @asjur !
> > Websocket feature is expected to be released towards the end of this year.
> 
> Any update on this?

Websocket no longer has an ETA. gRPC was now planned for start of next year, Websocket will follow sometime after that.

**Comment by DannyvanHolten** [comment]
Oef, use bruno seems like the perfect tool with saving the settings locally. But without websocket support it loses it's value for us.

**Comment by Vikaseneco** [comment]
@helloanoop When can we expect support for Web socket in Burno?

**Comment by anusree-bruno** [comment]
Hi! We're currently a bit packed as we're focused on the initial phase of gRPC support. Once that's released, we'll begin work on WebSocket integration.

**Comment by helloanoop** [comment]
🚀 Big News: WebSockets support has landed in Bruno v2.13.0! 🥳✨

<img width="1299" height="668" alt="Image" src="https://github.com/user-attachments/assets/74424e45-2b0c-45d6-9f43-0700f1831378" />

**Comment by gajanak** [comment]
Please Allow Binary Message - as hex ? if Message Display is possible as HexDump - it would be nice to allow Writing it same way.

Is script support for receiving Message planned ?

PS: Buy Golden Edition, too. Would buy it again - if it would be available. (Other Discussion.)

**Comment by helloanoop** [comment]
Hey @gajanak Do you mind creating a new issue around adding support binary message format and describe your use case and how this would help. Scripting support for websocket is planned
